### ISSUE TITLE
Incorrect evaluation of list comprehension

### ISSUE THREAD

**Issue by et2010** [abo-abo/lpy#17]
Hi, 

I found following evaluation gives wrong results:

```python
mylist = [1, 2, 3]
mylist2 = [
    x for x in mylist
    if True
]
```
After evaluation, `mylist2` is empty while the correct results should be `[1, 2, 3]`.

**Comment by abo-abo** [comment]
Can't reproduce on Linux. What's the result of evaluating `(lispy-eval-python-str)`?

**Comment by et2010** [comment]
> What's the result of evaluating (lispy-eval-python-str)?

```

"ne', color=colors[-1])"

```

@abo-abo

**Comment by abo-abo** [comment]
That doesn't look right. Can you move the point before `mylist2` and eval again?

**Comment by et2010** [comment]
Sure. 

```
mylist = [1, 2, 3]
mylist2 = [
           x for x in mylist
           if True
           ]
# => 


```

**Comment by abo-abo** [comment]
If I understood correctly, the second case works fine for both of us.

With the first case, I have `mylist2` equal to `[1, 2, 3]` although no output is produced. Is that also the case for you?

**Comment by et2010** [comment]
Yes, that's the case. But in a real-life scenario I've experienced, the evaluation failed and an incorrect value was assigned.
